python/cuml/dask/common/comms_utils.pyx

```diff
@@ -131,12 +135,10 @@ def inject_comms_on_handle(handle, nccl_inst, ucp_worker, eps, size, rank):
     """
     cdef size_t *ucp_eps = <size_t*> malloc(len(eps)*sizeof(size_t))
 
-    cdef void* ep_st
     for i in range(len(eps)):
         if eps[i] is not None:
-            ep_st = PyLong_AsVoidPtr(eps[i].get_ucp_endpoint())
-            pv = <size_t>&ep_st
-            ucp_eps[i] = pv
+            ep_st = <uintptr_t>eps[i].get_ucp_endpoint()
```

Comment:
The original cuml comms code did a cast to `ucp_ep_h` for this, however, this was removed because we don't explicitly build against ucp in the python/cython layer and our C++ uses `dlopen`. Do you know if it's somehow possible to import this from UCP?

Reply:
> Do you know if it's somehow possible to import this from UCP?

AFAIK, this isn't possible. In order to for Cython to cast the pointer it needs the type information at compile time.

Reply:
You can of cause include the definition of `ucp_ep_h` in your source but since it is an opaque pointer it would basically be the same as casting it to `<uintptr_t>`.